Data/ByteString/Lazy.hs

```diff
@@ -478,7 +476,7 @@ foldl' f z = go z
 -- (typically the right-identity of the operator), and a ByteString,
 -- reduces the ByteString using the binary operator, from right to left.
 foldr :: (Word8 -> a -> a) -> a -> ByteString -> a
-foldr k z cs = foldrChunks (flip (S.foldr k)) z cs
+foldr k = foldrChunks (flip (S.foldr k))
```

Comment:
These eta-contractions also have an effect on inlining behaviour. Mostly that's for the good, but not always. The rule with newer ghc is that it will inline only saturated calls, saturated based on the definition site. So here it will now inline if only the `k` is know, whereas before it would only inline if the `k` `z` and `cs` were known. In fact probably what we want for this example is to only inline if `k` and `z` are known.